implementations/micrometer-registry-new-relic/src/main/java/io/micrometer/newrelic/NewRelicMeterRegistry.java

```diff
@@ -134,7 +139,26 @@ public Builder threadFactory(ThreadFactory threadFactory) {
             return this;
         }
 
+        /**
+         * Note: This only has an effect when a {@link #clientProvider(NewRelicClientProvider)} is not provided and {@link NewRelicConfig#clientProviderType()} is {@link ClientProviderType#INSIGHTS_API}.
+         *
+         * @param httpClient http client to use for publishing
+         * @return builder
+         * @deprecated use {@link #clientProvider(NewRelicClientProvider)} instead.
+         */
+        @Deprecated
+        public Builder httpClient(HttpSender httpClient) {
+            this.httpClient = httpClient;
+            return this;
+        }
+
         public NewRelicMeterRegistry build() {
+            if (clientProvider == null && httpClient != null) {
+                //default to Insight API client provider if not specified in config or provided
+                clientProvider = (config.clientProviderType() == ClientProviderType.INSIGHTS_AGENT)
+                        ? new NewRelicInsightsAgentClientProvider(config)
+                        : new NewRelicInsightsApiClientProvider(config, httpClient, convention);
+            }
```

Comment:
Adding "&& httpClient != null" to condition won't allow config to specify Agent provider. The httpClient check should be in an inner condition that drives which API Client provider constructor to use.  An inner condition enables support for .builder(config).build(); and .builder(config).httpClient(httpClient).build(); for API publishing.

Reply:
I'm sure there's still some simplification possible, but I'm not understanding the current issue. Could you share a test that shows what's not working now? If the agent provider is configured in `NewRelicConfig`, then either 
A) the `clientProvider` will be set in this if block (if `httpClient` is also set - which won't be used for the agent provider)
or
B) `clientProvider` will remain `null` and the `NewRelicMeterRegistry` constructor will instantiate the `NewRelicInsightsAgentClientProvider`

Reply:
Ah, I wasn't taking into account the `NewRelicMeterRegistry` constructor behavior.  I was thinking that if `httpClient` was set, that the condition would be more clear if only the `NewRelicInsightsApiClientProvider` was used, from a pure backward compatibility perspective with Boot 2.2.x (config clientProviderType ignored).  Your approach also allows use of the Agent provider.